app/models/patient.rb

```diff
@@ -68,45 +68,41 @@ def latest_transfer
   # Patients who are eligible for reminders (exposure)
   scope :reminder_eligible_exposure, lambda {
     where(isolation: false)
+      .where(purged: false)
+      .where(public_health_action: 'None')
       .where(pause_notifications: false)
       .where('patients.id = patients.responder_id')
-      .where(purged: false)
-      .where.not(id: Patient.unscoped.exposure_under_investigation)
-      .where('last_date_of_exposure >= ? OR continuous_exposure = ?', ADMIN_OPTIONS['monitoring_period_days'].days.ago, true)
-      .left_outer_joins(:assessments)
-      .where_assoc_not_exists(:assessments, ['created_at >= ?', Time.now.getlocal('-04:00').beginning_of_day])
+      .where('last_date_of_exposure >= ? OR continuous_exposure = ?', (ADMIN_OPTIONS['monitoring_period_days'] + 1).days.ago, true)
+      .where.not('latest_assessment_at >= ?', Time.now.getlocal('-04:00').beginning_of_day)
       .or(
         where(isolation: false)
+          .where(purged: false)
           .where(pause_notifications: false)
+          .where(public_health_action: 'None')
           .where('patients.id = patients.responder_id')
-          .where(purged: false)
-          .where.not(id: Patient.unscoped.exposure_under_investigation)
-          .where('last_date_of_exposure >= ? OR continuous_exposure = ?', ADMIN_OPTIONS['monitoring_period_days'].days.ago, true)
-          .left_outer_joins(:assessments)
-          .where(assessments: { patient_id: nil })
+          .where('last_date_of_exposure >= ? OR continuous_exposure = ?', (ADMIN_OPTIONS['monitoring_period_days'] + 1).days.ago, true)
+          .where.not(latest_assessment_at: nil)
       )
       .distinct
   }
 
   # Patients who are eligible for reminders (isolation)
   scope :reminder_eligible_isolation, lambda {
     where(isolation: true)
+      .where(purged: false)
       .where(pause_notifications: false)
       .where('patients.id = patients.responder_id')
-      .where(purged: false)
       .where.not(id: Patient.unscoped.isolation_requiring_review)
       .where.not(id: Patient.unscoped.isolation_non_reporting_max)
-      .left_outer_joins(:assessments)
-      .where_assoc_not_exists(:assessments, ['created_at >= ?', Time.now.getlocal('-04:00').beginning_of_day])
+      .where.not('latest_assessment_at >= ?', Time.now.getlocal('-04:00').beginning_of_day)
       .or(
         where(isolation: true)
+          .where(purged: false)
           .where(pause_notifications: false)
           .where('patients.id = patients.responder_id')
-          .where(purged: false)
           .where.not(id: Patient.unscoped.isolation_requiring_review)
           .where.not(id: Patient.unscoped.isolation_non_reporting_max)
-          .left_outer_joins(:assessments)
-          .where(assessments: { patient_id: nil })
+          .where.not(latest_assessment_at: nil)
```

Comment:
This side of the OR should be for monitorees who never responded, but should still be reporting. Should this just be a `where` and not a `where.not`? (same for other scope)

Reply:
ah good catch, will make that change for the exposure scope as well and add tests for that case

Reply:
resolved here: https://github.com/SaraAlert/SaraAlert/pull/295/commits/a396b338a5c0f3dca839c29c986305ac8022632b#diff-9c324f49a7c201a8392c7c309c6477f2R103